src/location/mod.rs

```diff
@@ -303,6 +325,30 @@ impl ObPartitionInfo {
         &self.sub_part_desc
     }
 
+    pub fn part_tablet_id_map(&self) -> &HashMap<i64, i64> {
+        &self.part_tablet_id_map
+    }
+
+    pub fn get_partid_from_phyid(&self, phy_id: i64) -> i64 {
+        if ob_vsn_major() >= 4 {
+            let part_num = self
+                .sub_part_desc
+                .as_ref()
+                .map_or(0, |desc| desc.get_part_num());
+            let part_idx = extract_part_idx(phy_id) * part_num as i64 + extract_subpart_idx(phy_id);
+            self.part_tablet_id_map
+                .get(&part_idx)
+                .copied()
+                .unwrap_or_else(|| {
```

Comment:
Shall we return an error rather than panic if the part_idx is not found (as we all know, it's serious for a library to panic)?

Reply:
[unwrap_or_else](https://doc.rust-lang.org/core/?search=unwrap_or_else)
**unwrap_or_else** will not panic, it will execute the closure if there is None or Err in the result instead.